eda/src/autogluon/eda/analysis/shift.py

```diff
@@ -0,0 +1,347 @@
+from typing import Union, List, Any, Optional
+import warnings
+import pandas as pd
+import numpy as np
+from autogluon.core.metrics import balanced_accuracy, BINARY_METRICS
+from autogluon.tabular import TabularPredictor
+from .. import AnalysisState
+from .base import AbstractAnalysis
+
+
+def post_fit(func):
+    """decorator for post-fit methods"""
+
+    def pff_wrapper(self, *args, **kwargs):
+        assert self._is_fit, f'.fit needs to be called prior to .{func.__name__}'
+        return func(self, *args, **kwargs)
+
+    return pff_wrapper
+
+class Classifier2ST:
+    """A classifier 2 sample test, which tests for a difference between a source and target dataset.  It fits a
+    classifier to predict if a sample is in the source and target dataset, then computes an evaluation metric on a
+    holdout which becomes the test statistic.
+
+    Parameters
+    ----------
+    classifier_class : an AutoGluon predictor, such as autogluon.tabular.TabularPredictor
+        The predictor (classifier) class to classify the source from target dataset, predictor class needs to support
+        binary classification
+    sample_label : str, default = 'xshift_label'
+        The label that will be used to indicate if the sample is from training or test
+    eval_metric : callable, default = autogluon.core.metrics.balanced_accuracy
+        Binary classification metric to use for the classifier 2 sample test
+    split : float, default = 0.5
+        Training/test split proportion for classifier 2 sample test
+    classifier_kwargs : dict, default = {}
+        The kwargs passed to the classifier, a member of classifier_class
+    """
+    def __init__(self,
+                 classifier_class,
+                 sample_label='xshift_label',
+                 eval_metric=balanced_accuracy,
```

Comment:
Why not use `.predict_proba()`? IMO there is a big difference between very confident incorrect prediction (0.99999999 score for 0 while the label was 1) vs the one near threshold (0.50000001 score for 0 while label was 1) - this would be reflected differently in the `roc_auc` curve in ranking.